### PR TITLE
Rename ChatDecorator to MessageDecorator

### PR DESCRIPTION
**Pull request by apple502j**
#3197 redone. No javadocs removed except for inaccurate stuff.

### CODE REVIEW

File: mappings/net/minecraft/server/MinecraftServer.mapping [hunk 1]
```diff
@@ -414,10 +414,10 @@ CLASS net/minecraft/server/MinecraftServer
 	METHOD method_43825 setHostProfile (Lcom/mojang/authlib/GameProfile;)V
 		ARG 1 hostProfile
 	METHOD method_43928 shouldPreviewChat ()Z
-	METHOD method_43929 getChatDecorator ()Lnet/minecraft/class_7492;
-		COMMENT {@return the chat decorator used by the server}
+	METHOD method_43929 getMessageDecorator ()Lnet/minecraft/class_7492;
+		COMMENT {@return the message decorator used by the server}
 		COMMENT
-		COMMENT <p>See the documentation of {@link ChatDecorator} for more information.
+		COMMENT <p>See the documentation of {@link MessageDecorator} for more information.
```
Comment (liach) [inline mappings/net/minecraft/server/MinecraftServer.mapping]:
```suggestion
		COMMENT @see MessageDecorator
```

File: mappings/net/minecraft/network/MessageDecorator.mapping [hunk 2]
```diff
@@ -31,17 +32,16 @@ CLASS net/minecraft/class_7492 net/minecraft/network/ChatDecorator
...
 			COMMENT whether the decoration was previewed by the sender's client
 	METHOD decorateFiltered (Lnet/minecraft/class_3222;Lnet/minecraft/class_5837;)Ljava/util/concurrent/CompletableFuture;
-		COMMENT {@return the decorated signed chat message from undecorated {@code message}}
-		COMMENT
-		COMMENT <p>If {@code previewed} is false, the returned message will have the original
-		COMMENT content as signed and the decorated content as unsigned. This means that if the
-		COMMENT received player requires signed chat message, they will see the original content.
+		COMMENT {@return the decorated filtered message from undecorated {@code message}}
 		COMMENT
 		COMMENT <p>If {@code message} has a filtered part, this will decorate both the raw and the
```
Comment (liach) [inline mappings/net/minecraft/network/MessageDecorator.mapping]: I recommend a reword: The decorated message will be totally censored, partially filtered, or totally uncensored if and only if the original message is.

File: mappings/net/minecraft/network/MessageDecorator.mapping [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
-	COMMENT cache the result on preview, so that when the sent message needs decorating, the cached
-	COMMENT value can be used.
+	COMMENT and sent</strong>. If this is not followed correctly, the server will discard the message.
+	COMMENT For example, a decorator that appends the time the decoration was applied would be
+	COMMENT likely to fail, since there is usually a delay between the previewing and the
+	COMMENT submission. One way to solve this issue is to make it cache the result on preview,
+	COMMENT so that when the sent message needs decorating, the cached value can be used.
 	FIELD field_39384 NOOP Lnet/minecraft/class_7492;
-		COMMENT An empty chat decorator that does not decorate anything.
+		COMMENT An empty message decorator that does not decorate anything.
```
Comment (liach) [inline mappings/net/minecraft/network/MessageDecorator.mapping]:
```suggestion
		COMMENT An empty message decorator that returns the original message.
```

File: mappings/net/minecraft/network/MessageDecorator.mapping [hunk 4]
```diff
@@ -1,28 +1,29 @@
...
 	METHOD decorate (Lnet/minecraft/class_3222;Lnet/minecraft/class_2561;)Ljava/util/concurrent/CompletableFuture;
-		COMMENT {@return the signed chat message with unsigned content set as decorated {@code message}}
+		COMMENT {@return the decorated {@code message}}
 		COMMENT
-		COMMENT <p>If the received player requires signed chat message, they will see the original content.
+		COMMENT @apiNote Custom message decorators should implement this method (usually as
```
Comment (liach) [inline mappings/net/minecraft/network/MessageDecorator.mapping]: This note is quite useless as we see this is already an abstract method. I suggest removing this section.

File: mappings/net/minecraft/network/MessageDecorator.mapping [hunk 5]
```diff
@@ -1,28 +1,29 @@
...
-	COMMENT and sent</strong>. If this is not followed correctly, the server detects that the client
-	COMMENT sent a forged text and discards the message. For example, a decorator that appends the
-	COMMENT time the decoration was applied would be likely to fail, since there is usually a delay
-	COMMENT between the previewing and the submission. One way to solve this issue is to make it
-	COMMENT cache the result on preview, so that when the sent message needs decorating, the cached
-	COMMENT value can be used.
+	COMMENT and sent</strong>. If this is not followed correctly, the server will discard the message.
+	COMMENT For example, a decorator that appends the time the decoration was applied would be
+	COMMENT likely to fail, since there is usually a delay between the previewing and the
+	COMMENT submission. One way to solve this issue is to make it cache the result on preview,
```
Comment (liach) [inline mappings/net/minecraft/network/MessageDecorator.mapping]: I personally don't recommend caching; I rather recommend that the decorator return the same message given the same player instance and equal text objects, i.e. stay **idempotent** (and side-effect free)

**Review by liach — COMMENTED** [review]
Remember to update documentation in `SignedChatMessage`:
https://github.com/FabricMC/yarn/pull/3197/files#diff-edacd7ca77895ce4c62982bd29f0abe4c653cb98044986c34667d1854f7253c7

File: mappings/net/minecraft/network/MessageDecorator.mapping (hunk 5 above)
Comment (apple502j) [reply mappings/net/minecraft/network/MessageDecorator.mapping]: Yeah, caching is hard to implement (as I learned in my FAPI pr) - was unsure if we should keep it or not. Being idempotent works well, but the real usage doesn't require strict idempotence. I'll remove mentions to caching for now.

**Review by liach — APPROVED** [review]

**Review by haykam821 — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**